opentaskspal/src/main/java/org/dmfs/opentaskspal/readdata/PercentComplete.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.readdata;
+
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.Projection;
+import org.dmfs.android.contentpal.RowDataSnapshot;
+import org.dmfs.android.contentpal.projections.SingleColProjection;
+import org.dmfs.iterators.Function;
+import org.dmfs.optional.Optional;
+import org.dmfs.optional.decorators.DelegatingOptional;
+import org.dmfs.optional.decorators.Mapped;
+import org.dmfs.tasks.contract.TaskContract.Tasks;
+
+
+/**
+ * {@link Optional} for the {@link Tasks#PERCENT_COMPLETE} value of a task.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class PercentComplete extends DelegatingOptional<Integer>
+{
+    public static final Projection<Tasks> PROJECTION = new SingleColProjection<>(Tasks.PERCENT_COMPLETE);
+
+
+    public PercentComplete(@NonNull RowDataSnapshot<Tasks> task)
+    {
+        super(new Mapped<>(new Function<CharSequence, Integer>()
```

Comment:
I've noticed that pretty much all `DelegatingOptional` implementations which return a task value, delegate to a `Mapped` decorator with a Function and `rowDataSnapshot.charData(…)`. I think this could be extracted to a `OptionalTaskCharData` which would be used like so:

```
  super(new OptionalTaskCharData(columName, new Function(){…}));
```
This even works for `TaskTitle` if the function is the identity function.